### DISCUSSION TITLE
字体不一致问题

### DISCUSSION THREAD

**Discussion by PhotometricStereo** [sjtug/SJTUThesis#827] [提问] [answered]
问题描述：
去年使用了一个老版本（应该是2021年3月26号下载的版本）一直在写论文，昨天刚刚更新到overleaf上默认的[新版本](https://www.overleaf.com/)，发现了很多不一致的地方：

主要问题：
1. 字体不一致问题（如下图所示）
2. 行距变更问题（如下图所示）
左侧为老版本，字体和行距**正常**
右侧为新版本，字体变矮，行距变窄，标题也没有加粗了
![图片](https://user-images.githubusercontent.com/101957989/175821998-d0bb7518-519b-4ba7-80d3-70aa2d9aa9fc.png)![图片](https://user-images.githubusercontent.com/101957989/175821985-faa3d22f-6730-481b-96be-1deca0457e57.png)

已经参考过[这个问题](https://github.com/sjtug/SJTUThesis/discussions/800), 但问题是同样在overleaf上编译，如果是字体的问题，老版本也是用的默认字体为什么就能编译出正常的结果呢？

目前设置`\documentclass[type=doctor,twoside,zihao=-4]{sjtuthesis}`
想请教下，如果想修改得到左侧老版本的字体和行距，该修改哪些位置，非常感谢！

次要问题
1.在overleaf里的新版公式环境下\bm{}会报错。

非常感谢前辈们的解答！ @AlexaraWu @skyzh @LogCreative

**Comment by skyzh** [comment, marked as answer]
可以调整 cjk-font 文档属性，看起来原来的字体看起来是 ubuntu 或者 adobe，新版似乎自动用了 fandol 字体。

**Reply by LogCreative** [reply]
新版是操作系统检测，老版用的 fandol，新版用的 ubuntu。

**Reply by PhotometricStereo** [reply]
非常感谢您的解答！
现在尝试后发现字体依然不太work，ubuntu好像还是一样，同时overleaf上没有adobe~现在正在是fandol

**Reply by PhotometricStereo** [reply]
字体问题已经work了！！修改了`cjk-font`选项
使用`\documentclass[type=doctor,twoside,zihao=-4,cjk-font=fandol]{sjtuthesis}`解决~
左边老版本，右边新版本
![图片](https://user-images.githubusercontent.com/101957989/175824567-c58655a7-27d8-4e50-bd9d-b7de6daf90bf.png)![图片](https://user-images.githubusercontent.com/101957989/175824577-90cb7f04-9704-420c-aeed-b0b3e2a308f0.png)


另外想请教下全局行距的问题，测试了在`sjtuthesis.cls`文件下修改`linespread`
`\LoadClass[a4paper,UTF8,scheme=plain,fontset=none,linespread=1.3]{ctexbook}[2018/04/01]`
查了老版本是这样的：
`\LoadClass[a4paper,UTF8,scheme=plain,linespread=1.49]{ctexbook}[2018/04/01]`
左边老版本，右边新版本
![图片](https://user-images.githubusercontent.com/101957989/175824828-75690bf9-19c8-4a64-85d7-128b4245c84c.png)![图片](https://user-images.githubusercontent.com/101957989/175824830-d439db96-740b-4513-8ad4-ca74cd32dac1.png)


字体字号问题应该已经解决。最后一个问题就是标题看起来老版本可能粗点，但是两个版本的section定义看起了是一样的：
```
  section = {%
    format       = \zihao{4}\bfseries\heiti,
    afterindent  = true,
    afterskip    = {1ex \@plus .2ex},
  },
``

**Reply by PhotometricStereo** [reply]
补充请教一个问题无论是在overleaf上还是在sjtulatex都会不定时出现`! LaTeX Error: File  "sjtuthesis.cls" not found.`
overleaf：![图片](https://user-images.githubusercontent.com/101957989/175825089-d46598c3-782f-440a-94c3-94a3c3d76cbd.png)
这是由于编译器找不到相对路径嘛？

另外想请教下，`.cls`文件为什么要写成这样的路径呢：`./texmf/tex/latex/sjtuthesis/sjtuthesis.cls`
这个texmf是样式文件，看到`.latexmkrc`里面的宏定义`ensure_path( 'TEXINPUTS', './texmf//' );`-->是不是意味着latex可以在其中自己找到sjtuthesis这个cls呢？

**Reply by LogCreative** [reply]
如果真的要字体这么粗，将这一行换成
https://github.com/sjtug/SJTUThesis/blob/b164d0f5b3087ed86c1038b5a85014b47a92c6f2/texmf/tex/latex/sjtuthesis/fd/sjtu-cjk-font-fandol.def#L49
类似于 `fontset=windows` 的
https://github.com/sjtug/SJTUThesis/blob/b164d0f5b3087ed86c1038b5a85014b47a92c6f2/texmf/tex/latex/sjtuthesis/fd/sjtu-cjk-font-windows.def#L27

但请注意使用伪粗体后带来的 #813 问题。

**Reply by LogCreative** [reply]
> 另外想请教下，`.cls`文件为什么要写成这样的路径呢：`./texmf/tex/latex/sjtuthesis/sjtuthesis.cls` 这个texmf是样式文件，看到`.latexmkrc`里面的宏定义`ensure_path( 'TEXINPUTS', './texmf//' );`-->是不是意味着latex可以在其中自己找到sjtuthesis这个cls呢？

对的 latexmk 会在这个地方寻找依赖文件。在 Overleaf 上遇到这个问题，尝试在编译时在**主文档**按下编译按钮。

**Reply by PhotometricStereo** [reply]
非常感谢您的解答！

**Comment by LogCreative** [comment]
似乎 `\bm` 并没有问题，可以考虑贴这一部分的代码。

**Reply by PhotometricStereo** [reply]
一个实例，只要代码里面有`\bm`就编译不过去了：
work：`$\mathbf{L}·\mathbf{N}=|N||L|cos\alpha = cos \alpha$`
不work: `$\bm{L}·\bm{N}=|N||L|cos\alpha = cos \alpha$`，报错如下：
![图片](https://user-images.githubusercontent.com/101957989/175825934-06f3de74-e941-49b1-94ca-dc89888c7a78.png)

**Reply by LogCreative** [reply]
无法复现这个问题，检查是否引入了新的宏包（你可以在一个干净的文档中尝试这段代码），可以在底部 Other logs and files 中下载 output.log 文件拖至此框以查看全部的日志（如果数据不敏感）。

> 以及这段代码的标准写法
```latex
\bm{L}\cdot\bm{N}=\lvert N\rvert\lvert L\rvert\cos\alpha=\cos\alpha
```

**Reply by LogCreative** [reply]
也可以使用 `\boldsymbol` 来替代 `\bm`（见 https://tex.stackexchange.com/a/618468 ）。

**Comment by PhotometricStereo** [comment]
非常感谢！
